shell/platform/android/io/flutter/plugin/platform/SingleViewPresentation.java

```diff
@@ -235,14 +249,45 @@ private static int atMost(int measureSpec) {
         }
     }
 
+    /** Answers calls for {@link InputMethodManager} with an instance cached at creation time. */
+    private static class ImmContext extends ContextWrapper {
+        private @NonNull
+        final InputMethodManager inputMethodManager;
+
+        ImmContext(Context base) {
+            this(base, /*inputMethodManager=*/null);
+        }
+
+        private ImmContext(Context base, @Nullable InputMethodManager inputMethodManager) {
+            super(base);
+            this.inputMethodManager = inputMethodManager != null ? inputMethodManager : (InputMethodManager) base.getSystemService(INPUT_METHOD_SERVICE);
+        }
+
+        @Override
+        public Object getSystemService(String name) {
+            if (INPUT_METHOD_SERVICE.equals(name)) {
+                return inputMethodManager;
+            }
+            return super.getSystemService(name);
+        }
+
+        @Override
+        public Context createDisplayContext(Display display) {
+            Context displayContext = super.createDisplayContext(display);
+            return new ImmContext(displayContext, inputMethodManager);
+        }
+    }
+
     /**
-     * Proxies a Context replacing the WindowManager with our custom instance.
+     * Proxies a Context replacing the WindowManager and InputMethodManager with our custom instance.
      */
-    static class PresentationContext extends ContextWrapper {
-        private WindowManager windowManager;
-        private final WindowManagerHandler windowManagerHandler;
+    private static class PresentationContext extends ImmContext {
```

Comment:
I think I'd find it less confusing if this didn't extend ImmContext (the base context this class is given is already an ImmContext).

Reply:
Done.